torch/_higher_order_ops/base_hop.py

```diff
@@ -143,11 +174,20 @@ def gen_schema(self, *args, **kwargs):
             output,
         ) = check_input_alias_and_mutation_return_ouputs(subgraph, fake_args)
 
-        assert (
+        if not (
             len(inp_inp_alias) == 0
             and len(inp_out_alias) == 0
             and len(out_out_alias) == 0
-        ), "Aliasing is not suppported for HOP subgraph."
+        ):
+            # TODO: turn this into an error.
+            # test_foreach_map_backward_binary_foreach_map_addrecip_op fails the alias test.
+            import warnings
+
+            warnings.warn(
+                "Aliasing is not suppported for HOP subgraph.\n"
+                f"{subgraph.print_readable(print_output=False)}\n"
+                f"Alias info: inp-inp alias: {inp_inp_alias}, inp-out alias: {inp_out_alias}, out-out alias{out_out_alias}"
+            )
```

Comment:
say something like this may lead to silent incorrectness. Since this is silent incorrectness we should try to fix this asap (not in this PR)